### PR TITLE
Initial implementation of Jaccard similarity for vectors

### PR DESCRIPTION
**Pull request by pabloem**
Implementation of Jaccard vector similarity for positive, floating-point-valued vectors.

**Not implemented**
- Quantization

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo fmt` command prior to submission?
3. [x] Have you checked your code using `cargo clippy` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by pabloem** [comment]
@generall could you approve the CI to run? I am working on this this week : )

**Comment by pabloem** [comment]
Note passing tests on my fork for the same PR: https://github.com/pabloem/qdrant/pull/1

**Comment by pabloem** [comment]
@agourlay thanks for triggering test workflows. PTAL? : )

**Comment by agourlay** [comment]
Thanks for the contribution!

TBH we were unsure about supporting the Jaccard distance.

cc @generall

File: lib/segment/src/vector_storage/quantized/quantized_vectors.rs [hunk 1]
```diff
@@ -328,6 +328,8 @@ impl QuantizedVectors {
                 Distance::Cosine => quantization::DistanceType::Dot,
                 Distance::Euclid => quantization::DistanceType::L2,
                 Distance::Dot => quantization::DistanceType::Dot,
+                // TODO(pabloem): Figure out what sort of quantized distance to use for Jaccard.
+                Distance::Jaccard => quantization::DistanceType::Dot,
```
Comment (generall) [inline lib/segment/src/vector_storage/quantized/quantized_vectors.rs:332]: it would probably require a custom quantization distance

**Comment by generall** [comment]
Hi @pabloem, thanks for the contribution!

Unfortunately we can't enable features in the interface, which are only particularly implemented. Speaking about your `ToDo` comments regarding quantization support and other questions.

I would recommend to keep the implementation in this PR, but remove integration into the interface, so we can split the changes into multiple smaller PRs which are easier to digest.

Additionally, I would like to see more tests for the equivalency of SIMD and simple implementation of the distance

**Comment by generall** [comment]
We would like to re-iterate over jaccard distance in a separate task. We would plan step-by-step integration, so we shouldn't have large half-done PRs which changes the interface which we can't merge right away.

@pabloem, thanks for the attempt anyway!

**Pull request closed without merging**